File: hackerrank/30 days of code/day 20, sorting/day_20_sorting.ts
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8')

let inputString: string = ''
let inputLines: string[] = []
let currentLine: number = 0

process.stdin.on('data', function(inputStdin: string): void {
    inputString += inputStdin
});

process.stdin.on('end', function(): void {
    inputLines = inputString.split('\n')
    inputString = ''

    main()
});

function readLine(): string {
    return inputLines[currentLine++]
}

function bubbleSort(a: number[]) {
        
    let numberOfSwaps = 0

    for (let _ in a) {
        for (let j = 0, N = a.length - 1; j < N; j++)

            if (a[j] > a[j + 1]) {
                let temp = a[j]
                a[j] = a[j + 1]
                a[j + 1] = temp
                numberOfSwaps++
            }

        if (!numberOfSwaps)
            break
    }
    return numberOfSwaps
}

function main() {
    readLine()

    const list = readLine().replace(/\s+$/g, '').split(' ').map(temp => parseInt(temp, 10))

    console.log(`Array is sorted in ${bubbleSort(list)} swaps.`)
    console.log(`First Element: ${list[0]}`)
    console.log(`Last Element: ${list[list.length - 1]}`)
}

File: hackerrank/30 days of code/day 20, sorting/day_20_sorting.cs
using System.Collections.Generic;
using System.Linq;
using System;

class Solution
{
    static int bubbleSort(List<int> a)
    {        
        int numberOfSwaps = 0;

        for (int i = 0, N = a.Count; i < N; i++) {

            for (int j = 0, N2 = N - 1; j < N2; j++)

                if (a[j] > a[j + 1]) {
                    int temp = a[j];
                    a[j] = a[j + 1];
                    a[j + 1] = temp;
                    numberOfSwaps++;
                }

            if (numberOfSwaps == 0)
                break;
        }
        return numberOfSwaps;
    }

    static void Main(string[] args)
    {
        Console.ReadLine();

        List<int> list = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(value => Convert.ToInt32(value)).ToList();

        Console.WriteLine("Array is sorted in {0} swaps.", bubbleSort(list));
        Console.WriteLine("First Element: {0}", list.First());
        Console.WriteLine("Last Element: {0}", list.Last());
    }
}
